### PR TITLE
DisplayFusion: Add version 9.5

### PR DESCRIPTION
**Pull request by issaclin32**
**DisplayFusion** ([homepage](https://www.displayfusion.com/)) is a multi-monitor desktop managing tool.

Notes:
~* **DisplayFusion** shows as a tray icon while running. The uninstaller script avoids error when user tries to uninstall/update the package while DisplayFusion is still running in the background.~  (Use the installer instead.)

* I couldn't find any trackable source for minor version updates (e.g. 9.5 -> 9.5c). Therefore `autoupdate.hash` is not used in this manifest to avoid error.

### CODE REVIEW

**Comment by Ash258** [comment]
> DisplayFusion should be installed using the 'Download Installer Now' button above to work 100% correctly
> This no-installer version is ONLY for people interested in testing DisplayFusion

File: bucket/displayfusion.json [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    "homepage": "https://www.displayfusion.com/",
+    "license": {
+        "identifier": "Freeware",
+        "url": "https://www.displayfusion.com/License/"
+    },
+    "url": "https://www.binaryfortress.com/Data/Download/?package=displayfusion&log=101#/setup.exe",
+    "hash": "sha1:1548c2aafb8fbfc24e7a9d2b2d7ce07e40ae55a9",
+    "installer": {
+        "script": [
+            "Invoke-ExternalCommand \"$dir\\setup.exe\" -ArgumentList \"/verysilent /dir=$dir /noicons\" -RunAs",
```
Comment (Ash258) [inline bucket/displayfusion.json]:
```suggestion
            "Invoke-ExternalCommand \"$dir\\setup.exe\" -ArgumentList @('/verysilent', '/dir=\"$dir\"', '/noicons') -RunAs",
```

File: bucket/displayfusion.json [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+            "displayfusion.exe",
+            "DisplayFusion"
+        ]
+    ],
+    "checkver": {
+        "url": "https://www.displayfusion.com/Download/",
+        "regex": "Latest Version:</b> v([\\d.]+)"
+    },
+    "autoupdate": {
+        "url": "https://www.binaryfortress.com/Data/Download/?package=displayfusion&log=101#/setup.exe",
```
Comment (Ash258) [inline bucket/displayfusion.json]: Use direct URL

https://github.com/Ash258/scoop-Ash258/blob/cf7f909ed203145a36bd2f397b9b963d6988c0f5/bucket/DisplayFusion.json#L21

**Comment by issaclin32** [comment]
> > DisplayFusion should be installed using the 'Download Installer Now' button above to work 100% correctly
> > This no-installer version is ONLY for people interested in testing DisplayFusion

Thanks for noticing. I modified the manifest to run the installer (instead of using the portable version). It seems that DisplayFusion need to install some Windows services (e.g. `DisplayFusionServices`) to "work 100% correctly"

File: bucket/displayfusion.json [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    "homepage": "https://www.displayfusion.com/",
+    "license": {
+        "identifier": "Freeware",
+        "url": "https://www.displayfusion.com/License/"
+    },
+    "url": "https://binaryfortressdownloads.com/Download/BFSFiles/101/DisplayFusionSetup-9.5.exe#/setup.exe",
+    "hash": "sha1:1548c2aafb8fbfc24e7a9d2b2d7ce07e40ae55a9",
+    "installer": {
+        "script": [
+            "Invoke-ExternalCommand \"$dir\\$fname\" -ArgumentList @('/verysilent', '/dir=\"$dir\"', '/noicons') -RunAs",
```
Comment (Ash258) [inline bucket/displayfusion.json]: Sorry I am idiot.. 

```suggestion
            "Invoke-ExternalCommand \"$dir\\$fname\" -ArgumentList @('/verysilent', \"/dir=`\"$dir`\"\", '/noicons') -RunAs",
```

File: bucket/displayfusion.json [hunk 4]
```diff
@@ -0,0 +1,38 @@
+{
+    "version": "9.5",
+    "description": "Multi-monitor desktop managing tool",
+    "homepage": "https://www.displayfusion.com/",
+    "license": {
+        "identifier": "Freeware",
+        "url": "https://www.displayfusion.com/License/"
+    },
+    "url": "https://binaryfortressdownloads.com/Download/BFSFiles/101/DisplayFusionSetup-9.5.exe#/setup.exe",
```
Comment (Ash258) [inline bucket/displayfusion.json]: That is why you have `$fname`
```suggestion
    "url": "https://binaryfortressdownloads.com/Download/BFSFiles/101/DisplayFusionSetup-9.5.exe",
```

File: bucket/displayfusion.json [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+            "displayfusion.exe",
+            "DisplayFusion"
+        ]
+    ],
+    "checkver": {
+        "url": "https://www.displayfusion.com/Download/",
+        "regex": "Latest Version:</b> v([\\d.]+)"
+    },
+    "autoupdate": {
+        "url": "https://binaryfortressdownloads.com/Download/BFSFiles/101/DisplayFusionSetup-$version.exe#/setup.exe",
```
Comment (Ash258) [inline bucket/displayfusion.json]:
```suggestion
        "url": "https://binaryfortressdownloads.com/Download/BFSFiles/101/DisplayFusionSetup-$version.exe",
```

File: bucket/displayfusion.json (hunk 4 above)
Comment (issaclin32) [reply bucket/displayfusion.json]: Sorry. Didn't notice that.

**Comment by issaclin32** [comment]
I found that the hash value on `https://www.displayfusion.com/Download` is actually for version **9.5c**. 

The download link `https://www.binaryfortress.com/Data/Download/?package=displayfusion&log=101` also leads to `DisplayFusionSetup-9.5c.exe`

However, I can't find any page that contains the version number **9.5c**. The download page and update log remains at version 9.5.

**Comment by Ash258** [comment]
They are using their API with POST requests for version checking. 

image

**Comment by issaclin32** [comment]
> They are using their API with POST requests for version checking.

Thanks for the information. Is there any possible way to apply it to the manifest?

**Comment by Ash258** [comment]
It's response is mess. They encode version number using base64

image

**Comment by issaclin32** [comment]
I am closing this PR because it is outdated. I will open a new PR if needed.

**Pull request closed without merging**